src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -250,33 +250,30 @@ private void ClearCurrentBinding()
             this.VisualStateManager.ClearBoundProject();
         }
 
-        private void ApplyBindingInformation(BoundSonarQubeProject bound)
+        private void ApplyBindingInformation(BindingConfiguration bindingConfig)
         {
             // Set the project key that should become bound once the connection workflow has completed
-            this.VisualStateManager.BoundProjectKey = bound.ProjectKey;
-
-            // Recreate the connection information from what was persisted
-            ConnectionInformation connectionInformation = bound.CreateConnectionInformation();
+            this.VisualStateManager.BoundProjectKey = bindingConfig.Project.ProjectKey;
 
             Debug.Assert(this.ActiveSection != null, "Expected ActiveSection to be set");
             Debug.Assert(this.ActiveSection?.RefreshCommand != null, "Refresh command is not set");
             // Run the refresh workflow, passing the connection information
             var refreshCmd = this.ActiveSection.RefreshCommand;
-            if (refreshCmd.CanExecute(connectionInformation))
+            if (refreshCmd.CanExecute(bindingConfig))
             {
-                refreshCmd.Execute(connectionInformation); // start the workflow
+                refreshCmd.Execute(bindingConfig); // start the workflow
             }
         }
 
-        private BoundSonarQubeProject SafeReadBindingInformation()
+        private BindingConfiguration SafeGetBindingConfig()
         {
-            ISolutionBindingSerializer solutionBinding = this.GetService<ISolutionBindingSerializer>();
```

Comment:
The _ISolutionBindingSerializer_ was called from multiple locations before but this was the only one that wrapped the call in a try-catch block (I don't know if there was a specific bug that caused a crash here in the past). I've left it for now as it doesn't do any harm. However, it would be better if the new implementations of _IConfigurationProvider_ were "safe", rather than making all of the callers handle IO errors.